projects/plugins/jetpack/extensions/blocks/donations/amount/save.js

```diff
@@ -0,0 +1,47 @@
+/**
```

Comment:
I can still reproduce part of this issue. 

The block editor is being updated - the currencies change. The page is being updated - it shows the correct currency. The checkout lightbox is still out of date though.

https://d.pr/v/Z3mVHp

Not sure why it's happening - in the code view the amounts attributes are changed as well as the global attribute being changed.

Reply:
I can't seem to reproduce, is the code you are using up to date with the latest version of the patch?

https://user-images.githubusercontent.com/1989914/153840720-33df6ff9-064d-4458-a799-c6c44de5e477.mp4

Reply:
I can reproduce it with the latest changes. Looking at it with the dev console open, it looks as though it's a race condition.

If I save the post before the request to `/wpcom/v2/memberships/products` has completed then it will not update the currency used on the checkout screen.

I've just checked and this is reproducible on the master branch too, so I'll resolve this and open a new issue for it.

Reply:
Filed https://github.com/Automattic/jetpack/issues/22838

Reply:
This would also be linked to https://github.com/Automattic/jetpack/pull/22710#discussion_r804715693, I'll resolve this thread and we can continue the conversation on the issue you've opened.